api/src/main/java/com/epam/pipeline/manager/docker/scan/AggregatingToolScanManager.java

```diff
@@ -433,49 +444,62 @@ private DockerClient getDockerClient(String tool, DockerRegistry registry) {
         return dockerClientFactory.getDockerClient(registry, token);
     }
 
-    private ToolVersionScanResult convertResults(final ClairScanResult clairScanResult,
-                                                 final DockerComponentScanResult compScanResult,
-                                                 final Tool tool,
-                                                 final String tag,
-                                                 final String digest,
-                                                 final String defaultCmd,
-                                                 final int layersCount,
-                                                 final boolean hasCudnnVersion) {
+    private ToolVersionScanResult gatherResults(final Tool tool, final String tag,
+                                                final String lastLayerRef, final String digest,
+                                                final String defaultCmd, final int layersCount,
+                                                final boolean hasCudnnVersion) throws IOException {
+
+        ToolScanStatus toolScanStatus = ToolScanStatus.COMPLETED;
+
+        final Pair<Boolean, ClairScanResult> clairScanResult =
+                getScanResult(clairService != null, () -> clairService.getScanResult(lastLayerRef));
```

Comment:
It looks like the pair can be in two forms only: `(false, null)` and `(true, value)`. What do you think if we just use optional here? And same for `dockerComponentService`.

```suggestion
        final Optional<ClairScanResult> clairScanResult = Optional.ofNullable(clairService)
            .map(service -> service.getScanResult(lastLayerRef));
```

Reply:
good approach, thanks!